### PR TITLE
NEW: chatwoot integration

### LINKED ISSUE #301: Recieve feedback from customers + live customer chat
User should be able to provide feedback or ask doubts or questions / get live help while being on the platform

### PR DESCRIPTION
**Pull request by deepansh96**
Fixes #301 

## Summary

- Everytime a user logs in, a new thread will be created in the chatwoot app here https://plio-chatwoot.herokuapp.com/ with that user's email and name.
- As long as the user doesn't logout from the plio app, the user can access the past chats and any one of us can be assigned a user's chat and can talk to them through chatwoot app

TODO
- [x] update readme docs with the steps to set up chatwoot

## Test Plan

- [x] Test Responsiveness
    - [x] Laptop (1200px)
    - [x] Tablet (760px)
    - [x] Phone (320px)
- [x] Cross-Browser Testing
    - [x] Chrome
    - [x] Firefox
    - [x] Safari
- [x] Tested locally
- [x] Tested on staging
- [ ] Tested on production

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -32,3 +32,7 @@ VUE_APP_GOOGLE_ANALYTICS_ID=
 
 # mixpanel
 VUE_APP_MIXPANEL_PROJECT_TOKEN= # get this from the Mixpanel project dashboard
+
```
Comment (dalmia) [inline .env.example:35]: need to update `docs/ENV.md`

also, deployment would definitely fail - you need to update Dockerfile and both the staging and prod github .yml files.

File: src/App.vue [hunk 2]
```diff
@@ -146,6 +152,56 @@ export default {
     // https://vuex.vuejs.org/guide/state.html#object-spread-operator
     ...mapActions("auth", ["unsetAccessToken", "fetchAndUpdateUser"]),
     ...mapActions("sync", ["stopLoading"]),
+    mountChatwoot() {
+      // mounting chatwoot SDK to the DOM
+      let chatwootScript = document.createElement("script");
```
Comment (dalmia) [inline src/App.vue:157]: being more specific (script is too broad)
```suggestion
      let chatwootScript = document.createElement("chatwoot-script");
```

File: .env.example [hunk 3]
```diff
@@ -32,3 +32,7 @@ VUE_APP_GOOGLE_ANALYTICS_ID=
...
+# chatwoot
+VUE_APP_CHATWOOT_WEBSITE= # URL of the deployed chatwoot app
```
Comment (dalmia) [inline .env.example:37]: we have used URL everywhere.
```suggestion
VUE_APP_CHATWOOT_URL= # URL of the deployed chatwoot app
```

File: .env.example [hunk 4]
```diff
@@ -32,3 +32,7 @@ VUE_APP_GOOGLE_ANALYTICS_ID=
...
+VUE_APP_CHATWOOT_WEBSITE_TOKEN= # get this from your deployed chatwoot app
```
Comment (dalmia) [inline .env.example:38]:
```suggestion
VUE_APP_CHATWOOT_TOKEN= # get this from your deployed chatwoot app
```

File: src/App.vue (hunk 2 above)
Comment (deepansh96) [reply src/App.vue:157]: This is creating a `<script></script>` tag in HTML
There's nothing like a `<chatwoot-script></chatwoot-script>` tag in HTML

File: docs/ENV.md [hunk 5]
```diff
@@ -57,3 +57,10 @@ Name of the Google Analytics Property corresponding to the Tracking Code mention
 ### Mixpanel
 #### `VUE_APP_MIXPANEL_PROJECT_TOKEN`
 This is the project token for your mixpanel project which you can view in the Mixpanel Dashboard.
+
+### Chatwoot
```
Comment (dalmia) [inline docs/ENV.md:61]: would be good to add one saying what is Chatwoot used for.
I think would be good to do the same under the `Mixpanel` section too.

File: src/App.vue [hunk 6]
```diff
@@ -146,6 +152,56 @@ export default {
...
+      chatwootScript.innerHTML = `(function(d,t) {
```
Comment (dalmia) [inline src/App.vue:158]: fix postioning on phone

File: docs/ENV.md [hunk 7]
```diff
@@ -57,3 +57,10 @@ Name of the Google Analytics Property corresponding to the Tracking Code mention
...
+#### `VUE_APP_CHATWOOT_URL`
+The URL of your chatwoot app
+
+#### `VUE_APP_CHATWOOT_TOKEN`
+The token of your chatwoot app. Can be found in the app settings
```
Comment (dalmia) [inline docs/ENV.md:66]: tell that we need to pick this up from configuration code snippet

File: docs/ENV.md [hunk 8]
```diff
@@ -55,12 +55,18 @@ The Tracking Code for the Google analytics property corresponding to the app.
 Name of the Google Analytics Property corresponding to the Tracking Code mentioned above.
 
 ### Mixpanel
+Mixpanel is used for tracking user behaviour throughout the webapp.
+
 #### `VUE_APP_MIXPANEL_PROJECT_TOKEN`
 This is the project token for your mixpanel project which you can view in the Mixpanel Dashboard.
 
 ### Chatwoot
+Chatwoot is an open-source customer engagement suite. It's used to talk to customers and gather their feedback.
```
Comment (dalmia) [inline docs/ENV.md:64]:
```suggestion
Chatwoot is an open-source customer engagement suite. It is used to talk to customers and gather their feedback.
```

File: docs/crm/CHATWOOT.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
+## Chatwoot
+Plio uses [Chatwoot](https://github.com/chatwoot/chatwoot) as it's customer engagement suite. This document covers steps by step details on how to get the chatwoot app running and integrated into the frontend.
```
Comment (dalmia) [inline docs/crm/CHATWOOT.md:2]:
```suggestion
Plio uses [Chatwoot](https://github.com/chatwoot/chatwoot) as its customer engagement suite. This document covers steps by step details on how to get the chatwoot app running and integrated into the frontend.
```

**Pull request merged**